### PR TITLE
document accelerator=auto

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -289,6 +289,13 @@ if __name__ == "__main__":
     server.run(port=8000)
 ```
 
+The `accelerator` variable can be set to `"auto"`, and LitServe will select the appropriate GPU or CPU, whichever is available.
```
Comment (lantiga) [inline README.md:292]: `auto` is already the default, we should not instruct users to add it explicitly

So, remove it from the code, and instead add a section (under and expanded advanced bullet) that says: `accelerator` is `auto` by default.

If you wish to perform explicit placement, then pass the device manually + example.

**Review by lantiga — CHANGES_REQUESTED** [review]
Added a request

File: README.md [hunk 2]
```diff
@@ -226,12 +226,39 @@ class SimpleLitAPI(LitAPI):
...
+<details>
+    <summary>Automatic accelerator</summary>
+&nbsp;
+
+LitServe can automatically select the appropriate GPU or CPU, whichever is available on the user's machine.
+
+```python
+# Automatically selects the available accelerator
+
+server = LitServer(SimpleLitAPI())
```
Comment (williamFalcon) [inline README.md:244]:
```suggestion
api = SimpleLitAPI()

# these two are equivalent
server = LitServer(api)
server = LitServer(api, accelerator='auto')
```

File: README.md [hunk 3]
```diff
@@ -226,12 +226,43 @@ class SimpleLitAPI(LitAPI):
...
+LitServe can automatically select the appropriate GPU or CPU, whichever is available on the user's machine.
+
+```python
+# Automatically selects the available accelerator
+
+api = SimpleLitAPI()
+
+# these two are equivalent
+server = LitServer(api)
+server = LitServer(api, accelerator='auto')
```
Comment (williamFalcon) [inline README.md:248]: @lantiga @aniketmaurya 
this is how we should show things that happen automatically... an empty () doesn't show what this does correctly.

**Review by lantiga — APPROVED** [review]

**Pull request merged**